src/managers/EventManager.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import type { EventPublisher } from 'spark-protocol';
+import type { DeviceAttributeRepository, Event, EventData } from '../types';
+
+class EventManager {
+  _eventPublisher: EventPublisher;
+  _deviceAttributeRepository: DeviceAttributeRepository;
+
+  constructor(
+    deviceAttributeRepository: DeviceAttributeRepository,
+    eventPublisher: EventPublisher,
+  ) {
+    this._deviceAttributeRepository = deviceAttributeRepository;
+    this._eventPublisher = eventPublisher;
+  }
+
+  _filterEvents = (
+    eventHandler: (event: Event) => void,
+    userID?: string,
+    deviceID?: string,
+  ): (event: Event) => Promise<void> =>
+    async (event: Event): Promise<void> => {
```

Comment:
I'm just saying that resolve usually means that whatever you are doing succeeded and you usually reject the promise.

I understand what you are doing in this code and just using `return;` will make things clearer.